app/auth-auth0.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) Jupyter Development Team.
+ * Distributed under the terms of the Modified BSD License.
+ */
+/**
+ * Example: Passport JS strategy for Auth0.
+ * Requires additional configuration and npm packages. See
+ * https://github.com/jupyter-incubator/dashboards_server/wiki/Authentication
+ * for details.
+ */
+var config = require('./config');
+var passport = require('passport');
+var Auth0Strategy = require('passport-auth0');
+
+module.exports = function(app) {
+
+    app.get('/callback',
+        passport.authenticate('auth0', { failureRedirect: '/login' }),
+        function(req, res) {
+            if (!req.user) {
+                throw new Error('user null');
+            }
+            res.redirect("/");
+        }
+    );
+
+    app.get('/login',
+        passport.authenticate('auth0', {}), function (req, res) {
+            res.redirect("/");
+    });
+
+    app.post('/logout', function(req, res){
+        req.logout();
+        res.redirect('https://' + config.get('AUTH0_DOMAIN') + '/v2/logout?returnTo=' + config.get('PUBLIC_LINK_PATTERN') + '&client_id=' + config.get('AUTH0_CLIENT_ID'));
```

Comment:
I think it's better to construct URLs using an API rather than string concatenation. In particular, this works around issues where a user may or may not append a '/'.

In other parts of the code, we've used `url-join`, but not sure how well it handles URL params. If that doesn't work, consider using Node.js' [`url.format()`](https://nodejs.org/dist/latest-v4.x/docs/api/url.html#url_url_format_urlobj).